dds/DCPS/Service_Participant.cpp

```diff
@@ -97,6 +97,8 @@ void set_log_verbose(unsigned long verbose_logging)
 
 }
 
+extern char **environ;
```

Comment:
I didn't do an extensive search on this, but I'm not sure this is standard, as hinted by the warnings. Is there something in ACE we can use? If not, it's not pretty as it looks like it would have to be parsed, but there is [a Win32 call for this](https://docs.microsoft.com/en-us/windows/win32/api/processenv/nf-processenv-getenvironmentstrings?redirectedfrom=MSDN)

Reply:
It's not needed. The standard library has `getenv`.

Reply:
He could use `getenv` if he limited himself to predefined environment variables, but he's using `environ` to scan all the environment variables:

https://github.com/objectcomputing/OpenDDS/blob/5754111cff81e70bf2e48145aae59b051130def2/dds/DCPS/Service_Participant.cpp#L839-L850

Reply:
The only variables that can effect the program are those that match a predefined list.  That list would be a in a compile-time constant data structure that could be used to drive command line, ini file, and environment variable handling.

Reply:
I didn't see anything for this anymore on my review just now. Are you shelving it for now?

Reply:
Yes.  @mitza-oci  and I agreed that we can pursue that in a different PR.